src/foam/lib/json/Outputter.java

```diff
@@ -96,17 +97,26 @@ protected void outputNull() {
   }
 
   public void outputString(String s) {
+    boolean hasNewLine = s.indexOf('\n') >= 0;
+    if ( multiLineOutput_ && hasNewLine ) writer_.append("\r\n  ");
     writer_.append("\"");
+    if ( multiLineOutput_ && hasNewLine ) writer_.append("\"\"\r\n    ");
     writer_.append(escape(s));
+    if ( multiLineOutput_ && hasNewLine ) writer_.append("\r\n  \"\"");
     writer_.append("\"");
   }
 
   public String escape(String s) {
-    return s
-      .replace("\\", "\\\\")
+    boolean hasNewLine = s.indexOf('\n') >= 0;
+    s = s.replace("\\", "\\\\")
       .replace("\"", "\\\"")
       .replace("\t", "\\t")
       .replace("\n","\\n");
+    if ( multiLineOutput_ ) {
+      s = s.replace("\\n", "\r\n    ");
```

Comment:
We shouldn't turn line-feed in to carriage-return line feed.  Additionally if we're going to insert indentation in the outputter, we should ensure we also strip it in the parser, otherwise every time we update an object it will pick up additionall indentation.  Alternatively we should consider not adding extra indetation.